### PR TITLE
test: add transform tests to `v-once` tests

### PR DESCRIPTION
**Pull request by LittleSound**
refactor `vOnce.spec.ts` to use `compileWithOnce`

### CODE REVIEW

**Review by sxzz — COMMENTED** [review]
Please add complete tests from `packages/compiler-core/__tests__/transforms/vOnce.spec.ts` and mark them as `test.todo` or `test.fails`

**Comment by LittleSound** [comment]
OK, I have added more tests.

File: packages/compiler-vapor/__tests__/transforms/vOnce.spec.ts [hunk 1]
```diff
@@ -24,12 +42,118 @@ describe('v-once', () => {
         },
       },
     )
-    expect(code).matchSnapshot()
+    expect(ir.helpers.size).toBe(0)
+    expect(ir.effect).toMatchObject([])
```
Comment (sxzz) [inline packages/compiler-vapor/__tests__/transforms/vOnce.spec.ts:46]:
```suggestion
    expect(ir.effect).toEqual([])
```

File: packages/compiler-vapor/__tests__/transforms/vOnce.spec.ts [hunk 2]
```diff
@@ -24,12 +42,118 @@ describe('v-once', () => {
...
+
+    expect(code).toMatchSnapshot()
   })
 
   test('as root node', () => {
-    const code = compile(`<div :id="foo" v-once />`)
+    const { ir, code } = compileWithOnce(`<div :id="foo" v-once />`)
+
+    expect(ir.helpers.size).toBe(0)
+    expect(ir.effect).toMatchObject([])
```
Comment (sxzz) [inline packages/compiler-vapor/__tests__/transforms/vOnce.spec.ts:95]:
```suggestion
    expect(ir.effect).toEqual([])
```

File: packages/compiler-vapor/__tests__/transforms/vOnce.spec.ts [hunk 3]
```diff
@@ -24,12 +42,118 @@ describe('v-once', () => {
...
+    ])
+
     expect(code).toMatchSnapshot()
     expect(code).not.contains('effect')
   })
+
+  test('on nested plain element', () => {
+    const { ir, code } = compileWithOnce(`<div><div :id="foo" v-once /></div>`)
+    expect(ir.helpers.size).toBe(0)
+    expect(ir.effect).toMatchObject([])
```
Comment (sxzz) [inline packages/compiler-vapor/__tests__/transforms/vOnce.spec.ts:121]:
```suggestion
    expect(ir.effect).toEqual([])
```

**Pull request merged**